tests/phpunit/locallib_test.php

```diff
@@ -444,5 +444,46 @@ public function test_is_email_taken() {
         $this->assertFalse($auth->is_email_taken(strtoupper($user->email), $user->username));
         $this->assertFalse($auth->is_email_taken(ucfirst($user->email), $user->username));
     }
+    public function test_get_fallback_idp_returns_idp_object() {
+        global $DB, $saml2auth;
+        $this->resetAfterTest();
+        // First test when we have no idp's configured.
+        $fallbackidp = auth_saml2_get_fallback_idp();
+        $this->assertTrue(is_null($fallbackidp));
+
+        // Add two fake IdPs.
+        $metadataurl = 'https://idp.example.org/idp/shibboleth';
```

Comment:
Instead of inserting these, can you just set the config via the settings object so we are properly testing how it will be done for real?